code/modules/power/apc.dm

```diff
@@ -1279,11 +1279,34 @@ obj/machinery/power/apc/proc/autoset(var/val, var/on)
 	src.visible_message("<span class='notice'>[src]'s screen flickers with warnings briefly!</span>")
 	spawn(rand(2,5))
 		src.visible_message("<span class='notice'>[src]'s screen suddenly explodes in rain of sparks and small debris!</span>")
+		spark(get_turf(src), 8)
 		stat |= BROKEN
 		operating = 0
 		update_icon()
 		update()
 
+
+//Very violent failure caused by power overload. Dangerous to nearby personnel
+/obj/machinery/power/apc/proc/rupture()
+	var/turf/T = get_turf(src)
+	for (var/mob/living/L in range(5, src))
+		spark(get_turf(src), 8)

```

Comment:
...but there is an actual lightning effect.
Either way, 8 tiles of sparks _each_, plus the sparks that electrocute_act already creates is a lot of sparks. I doubt it will lag, but it's more sparks than anything else in the game currently creates.